app/src/main/java/org/mozilla/fenix/tabtray/TabTrayDialogFragment.kt

```diff
@@ -206,32 +213,33 @@ class TabTrayDialogFragment : AppCompatDialogFragment() {
         }
 
         // Check if this is the last tab of this session type
-        val isLastOpenTab = sessionManager.sessions.filter { snapshot.session.private }.size == 1
+        val isLastOpenTab = sessionManager.sessions.filter { snapshot.session.private == it.private }.size == 1
         val rootView = if (isLastOpenTab) { requireActivity().getRootView()!! } else { requireView().tabLayout }
         val anchorView = if (isLastOpenTab) { null } else { snackbarAnchor }
 
-        requireActivity().lifecycleScope.allowUndo(
-            rootView,
-            snackbarMessage,
-            getString(R.string.snackbar_deleted_undo),
-            {
-                sessionManager.add(snapshot.session, isSelected, engineSessionState = state)
-                _tabTrayView?.scrollToTab(snapshot.session.id)
-            },
-            operation = { },
-            elevation = ELEVATION,
-            paddedForBottomToolbar = isLastOpenTab,
-            anchorView = anchorView
-        )
-
-        dismissTabTrayIfNecessary()
+        if (isLastOpenTab) {
+            dismissTabTrayAndNavigateHome(sessionId)
+        } else {
+            requireActivity().lifecycleScope.allowUndo(
```

Comment:
Does this one have to be launched on the activity now since we aren't closing the tabs tray?